### PR TITLE
mathmod: init at 11.1-unstable-2024-01-26

### LINKED ISSUE #72469: Packaging requests: MathMod
**Project description**
Mathematical modelling software

**Metadata**

* homepage URL: https://www.facebook.com/parisolab/
* source URL: https://github.com/parisolab/mathmod
* license: gpl2
* platforms: cross-platform

### PR DESCRIPTION
**Pull request by TomaSajt**
## Description of changes

Closes https://github.com/NixOS/nixpkgs/issues/72469

This PR adds 1 package: `mathmod`

It seems to run alright

I am not copying the `.js` files, so some configuration features might not work properly, but I haven't found those yet.

The original project file wanted to copy all source files into a directory, so I removed that in a patch. The patch also sets the correct directory for the binary.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/science/math/mathmod/default.nix [hunk 1]
```diff
@@ -0,0 +1,35 @@
+{ lib
+, mkDerivation
+, fetchFromGitHub
+, substituteAll
+, qmake
+}:
+
+mkDerivation {
+  pname = "mathmod";
+  version = "12.0-dev-2023-09-07";
```
Comment (FliegendeWurst) [inline pkgs/applications/science/math/mathmod/default.nix:10]:
```suggestion
  version = "unstable-2023-09-07";
```

File: pkgs/applications/science/math/mathmod/default.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  postPatch = ''
+    substituteInPlace MathMod.pro --subst-var out
+  '';
+
+  nativeBuildInputs = [ qmake ];
+
+  meta = {
+    description = "A mathematical modelling software";
+    homepage = "https://github.com/parisolab/mathmod";
+    license = lib.licenses.gpl2Only;
```
Comment (FliegendeWurst) [inline pkgs/applications/science/math/mathmod/default.nix:29]:
```suggestion
    license = lib.licenses.gpl2Plus;
```
See e.g. https://github.com/parisolab/mathmod/blob/efcf329723cb1c5b836e1dd7edb76d67e81f5dbb/main.cpp#L5-L8

Comment (TomaSajt) [reply pkgs/applications/science/math/mathmod/default.nix]: Nice catch, thanks

**Review by FliegendeWurst — APPROVED** [review]
Tested some basic functionality, works fine.

Result of `nixpkgs-review pr 254623` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>mathmod</li>
  </ul>
</details>

**Pull request merged**